File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.java
class Solution {
    public long getDescentPeriods(int[] prices) {
        int i=0;
        int n=prices.length;
        long count=0;
        while(i<n)
        {
            int j=i+1;
            while(j<n && (prices[j-1]-prices[j])==1)
            {
                j++;
            }
            long l=j-i;
            count+=l*(l+1)/2;
            i=j;
        }
        return count;
    }
}

File: 2110-number-of-smooth-descent-periods-of-a-stock/2110-number-of-smooth-descent-periods-of-a-stock.py
class Solution:
    def getDescentPeriods(self, prices: list[int]) -> int:
        i=0
        n=len(prices)
        count=0
        while(i<n):
            j=i+1
            while(j<n and (prices[j-1]-prices[j])==1):
                j+=1
            l=j-i
            count+=l*(l+1)//2
            i=j
        return count